src/input/input_python.rs

```diff
@@ -240,12 +240,28 @@ impl<'a> Input<'a> for PyAny {
             // Safety: the gil is held while from_utf8 is running so py_byte_array is not mutated,
             // and we immediately copy the bytes into a new Python string
             let s = match from_utf8(unsafe { py_byte_array.as_bytes() }) {
-                // Why Python not Rust? to avoid an unnecessary allocation on the Rust side, the
+                // Why Python not Rust? to avoid an unnecessary allocation o(n the Rust side, the
                 // final output needs to be Python anyway.
                 Ok(s) => PyString::new(self.py(), s),
                 Err(_) => return Err(ValError::new(ErrorTypeDefaults::StringUnicode, self)),
             };
             Ok(s.into())
+        } else if coerce_numbers_to_str {
+            let py = self.py();
+            let decimal_type: Py<PyType> = get_decimal_type(py);
+
+            if self.is_instance_of::<PyInt>()
+                || self.is_instance_of::<PyFloat>()
+                || self.is_instance(decimal_type.as_ref(py)).unwrap_or_default()
```

Comment:
You can avoid rightwards drift and the duplication of the Err at the bottom by moving the is instance checks into a block expression:

```suggestion
        } else if coerce_numbers_to_str && {
            let py = self.py();
            let decimal_type: Py<PyType> = get_decimal_type(py);
            self.is_instance_of::<PyInt>()
                || self.is_instance_of::<PyFloat>()
                || self.is_instance(decimal_type.as_ref(py)).unwrap_or_default()
           }
```